MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -727,6 +727,14 @@
 #if MVK_MACOS
 	addSurfFmt(MTLPixelFormatRGB10A2Unorm);
 	addSurfFmt(MTLPixelFormatBGR10A2Unorm);
+#if (defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 101600)
```

Comment:
Ugh. In a pinch I threw similar guard code in to support `MTLLanguageVersion2_3`, but it's ugly and I wouldn't want it to propagate too far. And I guess when Xcode 12.2 comes out of beta, and becomes the standard, we can remove all of these.

If we don't think we can strip this out soon (if we need to support Xcode 12.0 or 11.7), and it might propagate even further (see other review note), it might be worth wrapping

     (defined(__MAC_OS_X_VERSION_MAX_ALLOWED) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 101600)

into something like a `MVK_MACOS_ARM` or `MVK_MACOS_APPLE_SILICON` macro?

That way, the intention would be clearer, and it would be easier to search and replace later.

Reply:
Agreed. That looks much cleaner. I also added one for the iOS conditional.

Reply:
Huh. So using the `defined` operator in a macro produces a warning, but not checking if the symbol is defined doesn't. Go figure.